sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
`latin-1` was just the encoding that I found that got the tests producing the correct values. I can try other encodings.

>shouldn't encoding/decoding happen even in py2?

The encoding I'm doing in these steps is specifically between bytes and str which are the same in py2 thus not requiring the encoding/decoding steps. The primary interface that makes these distinctions is working with the `struct` module I believe and so I'm just preprocessing the variables that go into and out of those functions.